src/site/content/en/blog/streams/index.md

```diff
@@ -0,0 +1,966 @@
+date: 2021-01-18
+# updated: 2021-01-18
+hero: hero.jpg
+alt: A forest stream with colored fallen leaves.
+tags:
+  - blog # blog is a required tag for the article to show up in the blog.
+  - capabilities
+  - streams
+---
+
+The Streams API allows you to programmatically access streams of data received over the network and
+process them with JavaScript. Streaming involves breaking a resource that you want to receive down
+into small chunks, and then processing these chunks bit by bit. While streaming is something
+browsers do anyway when receiving assets like HTML or videos to be shown on webpages, this
+capability has never been available to JavaScript before.
+
+Previously, if you wanted to process a resource of some kind (be it a video, or a text file, etc.),
+you would have to download the entire file, wait for it to be deserialized into a suitable format,
+and then process the whole lot after it is fully received. With streams being available to
+JavaScript, this all changes—you can now start processing raw data with JavaScript progressively as
+soon as it is available on the client-side, without needing to generate a buffer, string, or blob.
+This unlocks a number of use cases, some of which I list below:
+
+- **Video effects:** piping a readable video stream through a transform stream that applies effects
+  in real time.
+- **Data (de)compression:** piping a file stream through a transform stream that selectively
+  (de)compresses it.
+- **Image decoding:** piping an HTTP response stream through a transform stream that decodes bytes
+  into bitmap data, and then through another transform that translates bitmaps into PNGs. If
+  installed inside the `fetch` hook of a service worker, this allows you to transparently polyfill
+  new image formats like AVIF.
+
+## Core concepts
+
+Before I go into details on the various types of streams, let me introduce some core concepts first.
+
+### Chunks
+
+A chunk is a **single piece of data** that is written to or read from a stream. It can be of any
+type; streams can even contain chunks of different types. A chunk will often not be the most atomic
+unit of data for a given stream. For example, a byte stream might contain chunks consisting of 16
+KiB `Uint8Array` units, instead of single bytes.
+
+### Readable streams
+
+A readable stream represents a source of data, from which you can read. In other words, data **comes
+out** of a readable stream. Concretely, a readable stream is an instance of the `ReadableStream`
+class.
+
+### Writable streams
+
+A writable stream represents a destination for data, into which you can write. In other words, data
+**goes in** to a writable stream. Concretely, a writable stream is an instance of the
+`WritableStream` class.
+
+### Transform streams
+
+A transform stream consists of a **pair of streams**: a writable stream, known as its writable side,
+and a readable stream, known as its readable side. In a manner specific to the transform stream in
```

Comment:
```suggestion
and a readable stream, known as its readable side.
A real-world metaphor for this would be a
[simultaneous interpreter](https://en.wikipedia.org/wiki/Simultaneous_interpretation)
who on-the-fly translates from one language to another.
In a manner specific to the transform stream in
```